packages/dds/sequence/src/test/intervalCollection.spec.ts

```diff
@@ -1731,5 +1759,166 @@ describe("SharedString interval collections", () => {
 				assert.equal(collection.detachIndex(mockIntervalIndex), false);
 			});
 		});
+
+		describe("support querying intervals with endpoint in a range", () => {
+			let helpers: IIntervalHelpers<Interval>;
+			let endInRangeIndex: EndInRangeIndex<Interval>;
+			let compareFn;
+			let results;
+
+			beforeEach(() => {
+				helpers = {
+					compareEnds: (a: Interval, b: Interval) => a.end - b.end,
+					create: createInterval,
+				};
+				endInRangeIndex = new EndInRangeIndex(helpers);
+				// sort the query result by the interval endpoint value
+				compareFn = (a: Interval, b: Interval) => {
+					if (a.end === b.end) {
+						return a.start - b.start;
+					}
+					return a.end - b.end;
+				};
+			});
+
+			it("can not return result if the input does not meet specified requirement", () => {
```

Comment:
it's not clear to me what the intent of this test is--seems like it might be overloaded and thus difficult to name. I'd consider breaking it up into several tests for more clarity.

Reply:
potential interesting cases which could be individual tests:

- query range with start matching end
- intervals with negative endpoints